### PR TITLE
Update tdp spy tests for esm

### PR DESCRIPTION
**Pull request by wpears**
Mostly keeps test logic intact while replacing spies

### CODE REVIEW

File: test/unit_tests/apps/teachers-digital-platform/js/survey/grade-level-page-spec.js [hunk 1]
```diff
@@ -1,36 +1,50 @@
...
     document.body.innerHTML = HTML_SNIPPET;
   });
 
   it('should be recognized from HTML', () => {
-    const cookieSpy = jest.spyOn(Cookies, 'remove');
-    const modalSpy = jest.spyOn(modals, 'init');
     sessionStorage.setItem(ANSWERS_SESS_KEY, 'testItem');
 
-    surveys.init();
+    surveys.default.init();
```
Comment (anselmbradford) [inline test/unit_tests/apps/teachers-digital-platform/js/survey/grade-level-page-spec.js:43]: Maybe at the point of import have:

```
surveysModule = await import(
      '../../../../../../cfgov/unprocessed/apps/teachers-digital-platform/js/tdp-surveys.js'
    );
surveys = surveysModule.default;
```

That way we only need to touch the point of import when the import behavior changes in the future.

Comment (wpears) [reply test/unit_tests/apps/teachers-digital-platform/js/survey/grade-level-page-spec.js]: changed in 7e7c611

File: test/unit_tests/apps/teachers-digital-platform/js/survey/grade-level-page-spec.js [hunk 2]
```diff
@@ -1,36 +1,50 @@
...
 const HTML_SNIPPET = `
 <div data-tdp-page="grade-level">
 </div>
 `;
 
-xdescribe('The TDP survey grade-level page', () => {
-  beforeEach(() => {
+describe('The TDP survey grade-level page', () => {
+  let surveys;
+  const remove = jest.fn();
```
Comment (anselmbradford) [inline test/unit_tests/apps/teachers-digital-platform/js/survey/grade-level-page-spec.js:15]: To make the point where this is called a literal less generic, could this be named `cookieRemove`? Then in the unstable mock it'd be `remove: cookieRemove,` and the mock checks further down would be `expect(cookieRemove.mock.calls[0][0]).toEqual(RESULT_COOKIE);`.

File: test/unit_tests/apps/teachers-digital-platform/js/survey/results-page-spec.js [hunk 3]
```diff
@@ -1,55 +1,73 @@
...
-
 const $ = document.querySelector.bind(document);
 
-xdescribe('The TDP survey results page', () => {
+describe('The TDP survey results page', () => {
+  let initials, surveys, resultsPage, obfuscation;
+  const remove = jest.fn();
+  const init = jest.fn();
+  const close = jest.fn();
+  const copy = jest.fn();
```
Comment (anselmbradford) [inline test/unit_tests/apps/teachers-digital-platform/js/survey/results-page-spec.js:15]: Same as the `remove` comment, can we prefix these with what module they are in reference to, e.g.:

```
cookieRemove
modalsInit
modalsClose
clipboardCopy
```

Comment (wpears) [reply test/unit_tests/apps/teachers-digital-platform/js/survey/results-page-spec.js]: Done in 0522503

File: test/unit_tests/apps/teachers-digital-platform/js/survey/grade-level-page-spec.js (hunk 2 above)
Comment (wpears) [reply test/unit_tests/apps/teachers-digital-platform/js/survey/grade-level-page-spec.js]: Done in 0522503

File: test/unit_tests/apps/teachers-digital-platform/js/survey/survey-page-spec.js [hunk 4]
```diff
@@ -1,109 +1,63 @@
...
+    );
+
+    const surveyExports = await import(
+      '../../../../../../cfgov/unprocessed/apps/teachers-digital-platform/js/survey/survey-page.js'
+    );
+    scrollToEl = surveyExports.scrollToEl;
+    surveyPage = surveyExports.surveyPage;
+    surveys = await import(
+      '../../../../../../cfgov/unprocessed/apps/teachers-digital-platform/js/tdp-surveys.js'
+    );
```
Comment (anselmbradford) [inline test/unit_tests/apps/teachers-digital-platform/js/survey/survey-page-spec.js:33]: This one needs the `.default` treatment too.

File: test/unit_tests/apps/teachers-digital-platform/js/survey/survey-page-spec.js [hunk 5]
```diff
@@ -1,109 +1,63 @@
...
 } from '../../../../../../cfgov/unprocessed/apps/teachers-digital-platform/js/survey/config.js';
-import * as modals from '../../../../../../cfgov/unprocessed/apps/teachers-digital-platform/js/modals.js';
 import HTML_SNIPPET from '../../html/survey-page.js';
 
 const $ = document.querySelector.bind(document);
 
-xdescribe('The TDP survey page', () => {
+describe('The TDP survey page', () => {
+  const modalInit = jest.fn();
+  const close = jest.fn();
```
Comment (anselmbradford) [inline test/unit_tests/apps/teachers-digital-platform/js/survey/survey-page-spec.js:14]: Rename `modalClose`

File: test/unit_tests/apps/teachers-digital-platform/js/tdp-surveys-spec.js [hunk 6]
```diff
@@ -1,47 +1,58 @@
...
-  afterEach(() => {
-    jest.restoreAllMocks();
+const gl =
+  '../../../../../cfgov/unprocessed/apps/teachers-digital-platform/js/survey/grade-level-page.js';
+const res =
+  '../../../../../cfgov/unprocessed/apps/teachers-digital-platform/js/survey/result-page.js';
+const sur =
+  '../../../../../cfgov/unprocessed/apps/teachers-digital-platform/js/survey/survey-page.js';
+
+let surveys, g, r, s;
```
Comment (anselmbradford) [inline test/unit_tests/apps/teachers-digital-platform/js/tdp-surveys-spec.js:10]: Avoid single character variable names.

File: test/unit_tests/apps/teachers-digital-platform/js/survey/survey-page-spec.js [hunk 7]
```diff
@@ -1,109 +1,63 @@
...
   SCORES_UNSET_KEY,
 } from '../../../../../../cfgov/unprocessed/apps/teachers-digital-platform/js/survey/config.js';
-import * as modals from '../../../../../../cfgov/unprocessed/apps/teachers-digital-platform/js/modals.js';
 import HTML_SNIPPET from '../../html/survey-page.js';
 
 const $ = document.querySelector.bind(document);
 
-xdescribe('The TDP survey page', () => {
+describe('The TDP survey page', () => {
+  const modalInit = jest.fn();
```
[the remark this answers is not in the document]
Comment (wpears) [reply test/unit_tests/apps/teachers-digital-platform/js/survey/survey-page-spec.js]: e467290

File: test/unit_tests/apps/teachers-digital-platform/js/survey/survey-page-spec.js (hunk 5 above)
Comment (wpears) [reply test/unit_tests/apps/teachers-digital-platform/js/survey/survey-page-spec.js]: e467290

File: test/unit_tests/apps/teachers-digital-platform/js/survey/survey-page-spec.js (hunk 4 above)
Comment (wpears) [reply test/unit_tests/apps/teachers-digital-platform/js/survey/survey-page-spec.js]: e467290

**Review by anselmbradford — APPROVED** [review]

**Pull request merged**